docs/preview/features/message-pumps/customization.md

```diff
@@ -40,4 +40,34 @@ public class OrderMessagePump : MessagePump
 }
 ```
 
+## Control message context filtering
+
+When registering a new `IMessageHandler<>`, one can choose to add an additional filter on the message context used in that handler.
+
+Following example shows how a message handler should only process a certain message when a property's in the context is present.
+
+We'll use a simple message handler implementation:
+
+```csharp
+public class OrderMessageHandler : IMessageHandler<Order>
+{
+    public async Task ProcessMessageAsync(Order order, MessageContext context, ...)
```

Comment:
It's been a while so here is a stupid question - They were able to use different MessageContext type as well, right? Like the Service bus one?

Reply:
Not a stupid question 😉 , yest that's true, one can register message handlers with custom and built-in message context, altough I didn't implement or test this just yet in this PR. I thought it was a good start to first begin with the 'baseline' for this message context filtering.

Also why this PR is not closing the related issue 🤓 .

Reply:
Ok, thanks!

Reply:
But as explained below, I added some extensions for this too bc I needed it for the integration tests.